source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
@maxrudolph: Regarding "If you have a velocity field that contains piecewise rigid body motions everywhere in the domain, the correct solution is that the (dynamic) pressure is zero everywhere in the domain." -- that's not actually correct. Rather, the pressure needs to equal the gravitational potential plus an arbitrary constant, so that it counters the right hand side forces.

But you don't need to prescribe the pressure. If that's what the velocity field is, shouldn't the rest of the equations just happen to yield that pressure?